common/common/src/test/java/io/helidon/build/common/FileUtilsTest.java

```diff
@@ -0,0 +1,206 @@
+                OTHERS_EXECUTE);
+
+        Path zip = zip(outputDir.resolve("zip-permissions/archive.zip"), zipDir, path -> unchecked(() -> {
+            if (Files.isRegularFile(path)) {
+                Files.setPosixFilePermissions(path, permissions);
+            }
+        }));
+
+        readZipFileContent(zip, path -> unchecked(() -> {
+            Set<PosixFilePermission> posixPermissions = Files.getPosixFilePermissions(path);
+            assertThat(permissions, is(posixPermissions));
+        }));
+    }
+
+    @Test
+    @EnabledOnOs(OS.WINDOWS)
+    void testZipPermissions2() {
+        zip(outputDir.resolve("zip-permissions/archive.zip"), zipDir,
+                p -> unchecked(() -> Files.setPosixFilePermissions(p, Set.of(OWNER_EXECUTE))));
+    }
+
+    @Test
+    @EnabledOnOs({OS.LINUX, OS.MAC})
+    void testZipOriginalPermissions() throws IOException {
+        Set<PosixFilePermission> permissions = setPosixPermissions();
+        Path zip = zip(outputDir.resolve("original-permissions/archive.zip"), zipDir);
+
+        readZipFileContent(zip, path -> unchecked(() -> {
+            Set<PosixFilePermission> posixPermissions = Files.getPosixFilePermissions(path);
+            assertThat(permissions, is(posixPermissions));
+        }));
+    }
+
+    @Test
+    @EnabledOnOs(OS.WINDOWS)
+    void testZipOriginalPermissions2() {
+        zip(outputDir.resolve("original-permissions2/archive.zip"), zipDir);
+    }
+
+    @Test
+    @EnabledOnOs({OS.LINUX, OS.MAC})
+    void testUnzipPermissions() throws IOException {
+        Path directory = outputDir.resolve("unzip-permissions");
+        unzip(zipFile, directory);
+        Set<PosixFilePermission> permissions = Files.getPosixFilePermissions(directory.resolve("file"));
+        assertThat(permissions, is(Set.of(OWNER_READ, OWNER_EXECUTE)));
+    }
+
+    @Test
+    @EnabledOnOs(OS.WINDOWS)
+    void testUnzipPermissions2() {
+        Path directory = outputDir.resolve("unzip-permissions2");
+        unzip(zipFile, directory);
+        File file = directory.resolve("file").toFile();
+        assertThat(file.canRead(), is(true));
+        assertThat(file.canExecute(), is(true));
+    }
+
+    @Test
+    @EnabledOnOs({OS.LINUX, OS.MAC})
+    void testE2ePermissions() throws IOException {
+        Path directory = outputDir.resolve("e2e-permissions");
+        Set<PosixFilePermission> permissions = setPosixPermissions();
+        zipAndUnzip(directory);
+        list(directory).stream()
+                .filter(p -> !p.equals(outputDir) && Files.isRegularFile(p))
+                .forEach(path -> unchecked(() -> {
+                    Set<PosixFilePermission> posixPermissions = Files.getPosixFilePermissions(path);
+                    assertThat(permissions, is(posixPermissions));
+                }));
+    }
+
+    @Test
+    @EnabledOnOs(OS.WINDOWS)
+    void testE2ePermissions2() {
+        Path directory = outputDir.resolve("e2e-permissions2");
+        setPermissions();
```

Comment:
This is odd, and feels windows specific.
We only care about posix permissions, let's not assert anything.